src/scrabble.js

```diff
@@ -1,5 +1,165 @@
-function scrabble() {
-  // write code here
+const scoreLibrary = [
+  { letters: ['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T'], score: 1 },
+  { letters: ['D', 'G'], score: 2 },
+  { letters: ['B', 'C', 'M', 'P'], score: 3 },
+  { letters: ['F', 'H', 'V', 'W', 'Y'], score: 4 },
+  { letters: ['K'], score: 5 },
+  { letters: ['J', 'X'], score: 8 },
+  { letters: ['Q', 'Z'], score: 10 }
+]
+
+const scrabble = (str = '') => {
+  if (typeof str !== 'string') return 0
+
+  let scores = 0
+
+  const splitStr = str.split('')
+
+  if (
+    (splitStr.includes('{') && splitStr.includes('}')) ||
+    (splitStr.includes('[') && splitStr.includes(']')) ||
+    /^[A-Z]*$/.test(str.toUpperCase())
```

Comment:
Cool method used - what does the .test do?

Reply:
Same question here :)  Does it do something like only applying .toUpperCase() to a string if necessary?

Reply:
This method just check pattern which I used before calling this method. In my case this mathod just check if inside variable 'str' only characters if there is will be numbers of any another characters this test return false

Reply:
> Same question here :) Does it do something like only applying .toUpperCase() to a string if necessary?

Yes in this test I used just Capital letters and you should call .toUpperCase() before using method .test